pkg/profiler/cpu/maps.go

```diff
@@ -799,27 +818,56 @@ func (m *bpfMaps) setUnwindTableForMapping(buf *profiler.EfficientBuffer, pid in
 			return nil
 		}
 
-		threshold := min(uint64(len(ut)), m.availableEntries())
-		currentChunk := ut[:threshold]
-		restChunks := ut[threshold:]
-
-		numShards := len(ut) / maxUnwindTableSize
-		// The above numShards is correct if the unwind table we want to add
-		// snugly fits in the available space. We occupy one more shard if we
-		// don't have an empty live shard and if don't perfectly fit in the
-		// available space.
-		if m.highIndex%maxUnwindTableSize == 0 && uint64(len(ut)) > m.availableEntries() {
-			numShards++
-		}
+		chunkIndex := 0
 
-		// .len
-		if err := binary.Write(unwindShardsValBuf, m.byteOrder, uint64(numShards)); err != nil {
-			return fmt.Errorf("write shards .len bytes: %w", err)
-		}
```

Comment:
The previous logic was slightly incorrect, besides being quite repetitive. Now things should be improved!